docs/Implementing-easyblocks.rst

```diff
@@ -0,0 +1,486 @@
+.. _implementing_easyblocks:
+
+Implementing easyblocks
+=======================
+
+This documentation covers aspects of implementing *easyblocks* and how to use them.
+For a concise definition of easyblocks, see :ref:`easyblocks`.
+
+.. contents::
+    :depth: 3
+    :backlinks: none
+
+
+.. _implementing_easyblocks_basics:
+
+The basics
+----------
+
+An *easyblock* is a Python module that implements a software build and installation procedure.
+
+It can be considered a Python script that plugs into the :ref:`framework`.
+
+EasyBuild will leverage easyblocks as needed, depending on which software packages it needs to install.
+Which easyblock is required is determined by the ``easyblock`` easyconfig parameter, or the software name
```

Comment:
easyconfig parameter, if it is present, or the software name. In the latter case, EasyBuild will look for `<software>.py` in the easyblock include directories.